cmake/compile_definitions/linux.cmake

```
@@ -189,12 +189,29 @@ if(X11_FOUND)
             "${CMAKE_SOURCE_DIR}/src/platform/linux/x11grab.cpp")
 endif()
 
+# XDG portal
+if (${SUNSHINE_ENABLE_PORTAL})
+    pkg_check_modules(GIO gio-2.0 gio-unix-2.0)
+    pkg_check_modules(PIPEWIRE libpipewire-0.3)
+else()
+    set(GIO_FOUND OFF)
+    set(PIPEWIRE_FOUND OFF)
+endif()
+if(PIPEWIRE_FOUND)
+    add_compile_definitions(SUNSHINE_BUILD_PORTAL)
+    include_directories(${GIO_INCLUDE_DIRS} ${PIPEWIRE_INCLUDE_DIRS})
+    list(APPEND PLATFORM_LIBRARIES ${GIO_LIBRARIES} ${PIPEWIRE_LIBRARIES})
+    list(APPEND PLATFORM_TARGET_FILES
+            "${CMAKE_SOURCE_DIR}/src/platform/linux/portalgrab.cpp")
+endif()
+
 if(NOT ${CUDA_FOUND}
         AND NOT ${WAYLAND_FOUND}
         AND NOT ${X11_FOUND}
+        AND NOT ${PIPEWIRE_FOUND}
         AND NOT (${LIBDRM_FOUND} AND ${LIBCAP_FOUND})
         AND NOT ${LIBVA_FOUND})
-    message(FATAL_ERROR "Couldn't find either cuda, wayland, x11, (libdrm and libcap), or libva")
+    message(FATAL_ERROR "Couldn't find either cuda, wayland, x11, pipewire, (libdrm and libcap), or libva")
```

```suggestion
    message(FATAL_ERROR "Couldn't find either cuda, libva, pipewire, wayland, x11, or (libdrm and libcap)")
```

Let's alphabetize these.